src/main/java/org/truffleruby/core/regexp/RegexpNodes.java

```diff
@@ -459,12 +461,30 @@ private String formatError(DynamicObject string) {
             return StringUtils.format("invalid byte sequence in %s", Layouts.STRING.getRope(string).getEncoding());
         }
 
+        @Specialization(guards = {
+                "isInitialized(regexp)",
+                "isRubyString(string)",
```

Comment:
This should have a `"isValidEncoding(string, rangeNode)"` guard so it's mutually exclusive with `searchRegionInvalidEncoding` above.